packages/editor-ui/src/components/InviteUsersModal.vue

```diff
@@ -12,252 +11,181 @@ import {
+		{
+			name: 'role',
+			initialValue: ROLE.Member,
+			properties: {
+				label: i18n.baseText('auth.role'),
+				required: true,
+				type: 'select',
+				options: [
 					{
-						interpolate: { count: this.emailsCount.toString() },
+						value: ROLE.Member,
+						label: i18n.baseText('auth.roles.member'),
 					},
-				);
-			}
-
-			return this.$locale.baseText(
-				`settings.users.inviteUser${this.settingsStore.isSmtpSetup ? '' : '.inviteUrl'}`,
-			);
-		},
-		enabledButton(): boolean {
-			return this.emailsCount >= 1;
-		},
-		invitedUsers(): IUser[] {
-			return this.showInviteUrls
-				? this.usersStore.allUsers.filter((user) =>
-						this.showInviteUrls!.find((invite) => invite.user.id === user.id),
-					)
-				: [];
-		},
-		isAdvancedPermissionsEnabled(): boolean {
-			return this.settingsStore.isEnterpriseFeatureEnabled[
-				EnterpriseEditionFeature.AdvancedPermissions
-			];
+					{
+						value: ROLE.Admin,
+						label: i18n.baseText('auth.roles.admin'),
+						disabled: !isAdvancedPermissionsEnabled.value,
+					},
+				],
+				capitalize: true,
+			},
 		},
-	},
-	methods: {
-		validateEmails(value: string | number | boolean | null | undefined) {
-			if (typeof value !== 'string') {
-				return false;
-			}
+	];
+});
 
-			const emails = value.split(',');
-			for (let i = 0; i < emails.length; i++) {
-				const email = emails[i];
-				const parsed = getEmail(email);
+const emailsCount = computed((): number => {
+	return emails.value.split(',').filter((email: string) => !!email.trim()).length;
+});
 
-				if (!!parsed.trim() && !VALID_EMAIL_REGEX.test(String(parsed).trim().toLowerCase())) {
-					return {
-						messageKey: 'settings.users.invalidEmailError',
-						options: { interpolate: { email: parsed } },
-					};
-				}
-			}
+const buttonLabel = computed((): string => {
+	if (emailsCount.value > 1) {
+		return i18n.baseText(
+			`settings.users.inviteXUser${settingsStore.isSmtpSetup ? '' : '.inviteUrl'}`,
+			{
+				interpolate: { count: emailsCount.value.toString() },
+			},
+		);
+	}
 
-			return false;
-		},
-		onInput(e: { name: string; value: InvitableRoleName }) {
-			if (e.name === 'emails') {
-				this.emails = e.value;
-			}
-			if (e.name === 'role') {
-				this.role = e.value;
-			}
-		},
-		async onSubmit() {
-			try {
-				this.loading = true;
+	return i18n.baseText(`settings.users.inviteUser${settingsStore.isSmtpSetup ? '' : '.inviteUrl'}`);
+});
+const enabledButton = computed((): boolean => {
+	return emailsCount.value >= 1;
+});
+const invitedUsers = computed((): IUser[] => {
+	return showInviteUrls.value
+		? usersStore.allUsers.filter((user) =>
+				showInviteUrls.value?.find((invite) => invite.user.id === user.id),
+			)
+		: [];
+});
+const isAdvancedPermissionsEnabled = computed((): boolean => {
+	return settingsStore.isEnterpriseFeatureEnabled[EnterpriseEditionFeature.AdvancedPermissions];
+});
+
+function validateEmails(value: string | number | boolean | null | undefined) {
```

Comment:
I prefer normal functions because they are hoisted in normal JS context, so I can correctly use them in a computed function above it. If it's a variable, then it feels weird/should not be correct to use it in a computed function above for example. 

that being said, I tested it and there's no lint error or type error. Not sure why it works with variables but it does 🤷🏽